docs/contributing.rst

```diff
@@ -12,3 +12,44 @@ without any warning.
 
 .. include:: ../.github/CONTRIBUTING.rst
    :start-after: DO-NOT-REMOVE-deps-snippet-PLACEHOLDER
+
+Adding a new rule
+-----------------
+
+Writing a new rule became easier as we now allow you to include
+**implementation, testing and documentation** in a single rule file.
+
+One good example is MetaTagValidRule_ which can easily be copied in order
+to create a new rule by following the steps below:
+
+* Install test dependencies using ``pip install -r test-requirements.txt``. You
+  may want to use a :doc:`virtual environment <python3:tutorial/venv>` for
+  that.
+* Use a short by clear class name, which must match the filename
+* Pick an unused ``id``, the first number is used to determine its category,
+  see which one fits best.
+* Be sure ``experimental`` tag is included. Any new rule must stay as
+  experimental for at least two weeks until this tag is removed in next major
+  release.
+* Update all class level variables
+* Implement linting methods needed by you. Feel free to look on how similar
+  rules were implemented.
+* Update the tests. It must have at least one test and likely also a negative
+  match one. If the rule is task specific, you may want to test its use inside
+  blocks as well.
```

Comment:
Suggested rephrase: If the rule is task specific, it may be best to include a test to verify its use inside blocks as well.